libs/full/async_cuda/src/cuda_future.cpp

```diff
@@ -265,9 +269,22 @@ namespace hpx { namespace cuda { namespace experimental { namespace detail {
     // -------------------------------------------------------------
     void unregister_polling(hpx::threads::thread_pool_base& pool)
     {
+#if defined(HPX_DEBUG)
+        {
+            std::unique_lock<hpx::cuda::experimental::detail::mutex_type> lk(
+                detail::get_list_mtx());
+            HPX_ASSERT_MSG(get_event_queue().size_approx() == 0,
```

Comment:
We shouldn't have locks active when calling ASSERTs as this will cause the lock registration to kick in otherwise.

Reply:
Good catch! I've updated this to avoid that.